rust/ql/lib/codeql/rust/frameworks/http.model.yml

```
@@ -0,0 +1,13 @@
+extensions:
+  - addsTo:
+      pack: codeql/rust-all
+      extensible: sourceModel
+    data:
+      - ["repo:https://github.com/hyperium/hyper:hyper", "<crate::client::conn::http1::SendRequest>::send_request", "ReturnValue", "remote", "manual"]
```

I _think_ that using the new `Future` content type might fix the false negative [on this line](https://github.com/github/codeql/pull/18671/files#diff-86d492388ba1c06d122501572d4f442c46204e34eac64263c23bb9b30ed19b00R109).
```suggestion
      - ["repo:https://github.com/hyperium/hyper:hyper", "<crate::client::conn::http1::SendRequest>::send_request", "ReturnValue.Future.Variant[crate::result::Result::Ok(0)]", "remote", "manual"]
```

---

Ooh, I'll try it...

---

I've accepted the commit, and merged `main` since it requires a couple of other recently merged PRs to work (sortof).